eqcorrscan/core/template_gen.py

```diff
@@ -807,15 +815,13 @@ def _template_gen(picks, st, length, swin='all', prepick=0.05,
             used_tr = True
         if not used_tr:
             Logger.warning('No pick for {0}'.format(tr.id))
-    if plot and len(st1) > 0:
-        fig1 = tplot(st1, background=stplot, picks=picks_copy,
-                     title='Template for ' + str(st1[0].stats.starttime),
-                     show=False, return_figure=True)
-        fig2 = noise_plot(
-            signal=st1, noise=noise, show=False, return_figure=True)
-        fig1.savefig("{0}_template.png".format(st1[0].stats.starttime))
-        fig2.savefig("{0}_noise.png".format(st1[0].stats.starttime))
-        del(stplot, fig1, fig2)
+    if plot and plotdir and len(st1) > 0:
+        tplot(st1, background=stplot, picks=picks_copy, plotdir=plotdir)
+        noise_plot(signal=st1, noise=noise, plotdir=plotdir)
+    elif plot and len(st1) > 0:
+        tplot(st1, background=stplot, picks=picks_copy, show=True)
+        noise_plot(signal=st1, noise=noise, show=True)
+    del(stplot)
```

Comment:
This breaks the tests - `stplot` only exists if `plot=True`. Can you change this to something like:
```python
if plot:
    plot_kwargs = dict(show=True, save=False)
    if plotdir:
        if not os.path.isdir(plotdir):
            os.makedirs(plotdir)
        plot_kwargs.update({"save": True, "show": False})
    plot_kwargs.update(
        {"savefile": "{0}/{1}_SNR.png".format(plotdir, st1[0].stats.starttime),
         "title": "SNR for {0}".format(st1[0].stats.starttime)})
    noise_plot(signal=st1, noise=noise, **plot_kwargs)
    plot_kwargs.update(
        {"savefile": "{0}/{1_template.png}".format(plotdir, template[0].stats.starttime),
         "title": "SNR for {0}".format(st1[0].stats.starttime)})
    tplot(st1, background=stplot, picks=picks_copy, **plot_kwargs)
    del(stplot)
```